diag_manager/fms_diag_file_object.F90

```diff
@@ -1005,18 +1009,18 @@ subroutine add_start_time(this, start_time)
   class(fmsDiagFile_type), intent(inout)       :: this           !< The file object
   TYPE(time_type),         intent(in)          :: start_time     !< Start time to add to the fileobj
 
-  !< If the start_time sent in is equal to the base_time return because
-  !! this%start_time was already set to the base_time
-  if (start_time .eq. get_base_time()) return
+  !< If the start_time sent in is equal to the diag_init_time return because
+  !! this%start_time was already set to the diag_init_time
+  if (start_time .eq. diag_init_time) return
 
-  if (this%start_time .ne. get_base_time()) then
-    !> If the this%start_time is not equal to the base_time from the diag_table
+  if (this%start_time .ne. diag_init_time) then
+    !> If the this%start_time is not equal to the diag_init_time from the diag_table
     !! this%start_time was already updated so make sure it is the same or error out
     if (this%start_time .ne. start_time)&
       call mpp_error(FATAL, "The variables associated with the file:"//this%get_file_fname()//" have"&
       &" different start_time")
   else
-    !> If the this%start_time is equal to the base_time,
+    !> If the this%start_time is equal to the diag_init_time,
```

Comment:
Is the code below here needed?  Your comments at LL1012-1014 indicate it will never be executed because start_time=this%start_time and the early return at 1014 will prevail.

Reply:
Note that  the `start_time` is the time passed into the `register_diag_field` call. 

if the diag_time_init is not passed into diag_manager_init, `this%star_time` and `diag_init_time` will be set to the base date in the beginning of the run. 

In the second year of the simulation, `start_time` won't be equal to `diag_init_time`, so it will get passed L1012-1014.

 this%start_time will be equal to `diag_init_time`, so it will get to the else and this%start_time and the other variables will be set to start_time. This will only happen for the first variable in the file, after that `start_time` is expected to the same this%start_time or you will get to the FATAL.